test/integration/network_reserved_ips_test.go

```diff
@@ -0,0 +1,284 @@
+package integration
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	. "github.com/linode/linodego"
+)
+
+// TestReservedIPAddresses_InsufficientPermissions tests the behavior when a user account
+// doesn't have the can_reserve_ip flag enabled
+func TestReservedIPAddresses_InsufficientPermissions(t *testing.T) {
```

Comment:
Part of the test checks for insufficient permission case (reserve / delete) while the rest of the cases covered here are sanity checks that could probably be moved to another test case.

Reply:
I am pretty sure I checked this, but last I remember we needed permission to list and fetch reserved IP's as well.

Reply:
I can confirm that and change it if we do not need permission for them.

Reply:
My comment was based on the requirement for the `can_reserve_ip ` which only applies to reserve/unreserve.

Iit seems like you're checking for general permissions to use the feature, which is a different flag that would apply to all new endpoints, but shouldn't matter for this test case. I would just update the comment to reflect `permission to use the reservedIP feature`

Reply:
Yeah that makes more sense. I will be changing the comment.